src/lib/storage/table.hpp

```diff
@@ -76,7 +77,20 @@ class Table : private Noncopyable {
   // note this is slow and not thread-safe and should be used for testing purposes only
   void append(std::vector<AllTypeVariant> values);
 
+  // compresses a ValueColumn into a DictionaryColumn
+  void compress_chunk(ChunkID chunk_id);
+
  protected:
-  // Implementation goes here
+  uint32_t _max_chunk_size;
```

Comment:
this could be const

Reply:
True in theory. Creates some build errors for me though and I'm not willing to investigate.

Won't Fix.

Reply:
Can relate, the compiler can be very moody